src/hotspot/share/memory/heapInspection.hpp

```diff
@@ -245,7 +245,7 @@ class ParHeapInspectTask : public AbstractGangTask {
       _filter(filter),
       _missed_count(0),
       _success(true),
-      _mutex(Mutex::leaf, "Parallel heap iteration data merge lock") {}
+      _mutex(Mutex::leaf, "Parallel heap iteration data merge lock", Mutex::_safepoint_check_always) {}
```

Comment:
Is this lock even used by JavaThreads. ??

Reply:
It doesn't look like this is.  If we change the _allow_vm_block check in Mutex::check_block_state to be:

     if (!_enforces_nsv && (!thread->isJavaThread() && is_at_safepoint())) {
         // some good explanatory comment
         fatal("A NonJava thread could block on lock that may be held by a JavaThread during safepoint: %s", name());
    }

I think this lock will get caught by this assert.